panoramapublic/src/org/labkey/panoramapublic/PanoramaPublicFileImporter.java

```diff
@@ -0,0 +1,171 @@
+package org.labkey.panoramapublic;
+
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.Nullable;
+import org.labkey.api.admin.AbstractFolderImportFactory;
+import org.labkey.api.admin.FolderImportContext;
+import org.labkey.api.admin.FolderImporter;
+import org.labkey.api.admin.ImportException;
+import org.labkey.api.admin.SubfolderWriter;
+import org.labkey.api.data.Container;
+import org.labkey.api.exp.api.ExpData;
+import org.labkey.api.exp.api.ExpRun;
+import org.labkey.api.exp.api.ExperimentService;
+import org.labkey.api.files.FileContentService;
+import org.labkey.api.pipeline.PipelineJob;
+import org.labkey.api.pipeline.PipelineService;
+import org.labkey.api.query.BatchValidationException;
+import org.labkey.api.security.User;
+import org.labkey.api.writer.VirtualFile;
+import org.labkey.panoramapublic.pipeline.CopyExperimentPipelineJob;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This importer does a file move instead of copy to the temp directory and creates a symlink in place of the original
+ * file.
+ */
+public class PanoramaPublicFileImporter implements FolderImporter
+{
+    @Override
+    public String getDataType()
+    {
+        return PanoramaPublicManager.PANORAMA_PUBLIC_FILES;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return null;
+    }
+
+    @Override
+    public void process(@Nullable PipelineJob job, FolderImportContext ctx, VirtualFile root) throws Exception
+    {
+        Logger log = ctx.getLogger();
+
+        FileContentService fcs = FileContentService.get();
+        if (null == fcs)
+            return;
+
+        File targetRoot = fcs.getFileRoot(ctx.getContainer());
+
+        if (null == targetRoot)
+        {
+            log.error("File copy target folder not found: " + ctx.getContainer().getPath());
+            return;
+        }
+
+        if (null == job)
+        {
+            log.error("Pipeline job not found.");
+            return;
+        }
+
+        if (job instanceof CopyExperimentPipelineJob expJob)
```

Comment:
Maybe we should log if it's not a CopyExperimentPipelineJob?

Reply:
added